core/src/main/java/nl/nn/adapterframework/doc/model/FrankElement.java

```diff
@@ -1,27 +1,231 @@
+/* 
+Copyright 2020 WeAreFrank! 
+
+Licensed under the Apache License, Version 2.0 (the "License"); 
+you may not use this file except in compliance with the License. 
+You may obtain a copy of the License at 
+
+    http://www.apache.org/licenses/LICENSE-2.0 
+
+Unless required by applicable law or agreed to in writing, software 
+distributed under the License is distributed on an "AS IS" BASIS, 
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+See the License for the specific language governing permissions and 
+limitations under the License. 
+*/
+
 package nl.nn.adapterframework.doc.model;
 
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
 import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import org.apache.logging.log4j.Logger;
 
 import lombok.Getter;
-import lombok.Setter;
+import nl.nn.adapterframework.doc.DocWriterNew;
+import nl.nn.adapterframework.doc.Utils;
+import nl.nn.adapterframework.util.LogUtil;
 
 public class FrankElement {
+	private static Logger log = LogUtil.getLogger(FrankElement.class);
+
 	private final @Getter String fullName;
 	private final @Getter String simpleName;
-	private @Getter @Setter FrankElement parent;
-	private @Getter @Setter List<FrankAttribute> attributes;
-	private @Getter @Setter List<ConfigChild> configChildren;
+	private final @Getter boolean isAbstract;
+
+	// Represents the Java superclass.
+	private @Getter FrankElement parent;
+
+	private LinkedHashMap<String, FrankAttribute> attributes;
+	private LinkedHashMap<ConfigChild.Key, ConfigChild> configChildren;
```

Comment:
Als je in ElementChild een Kind maakt, zoals:

	enum Kind {
		ATTRIBUTE,
		CONFIGCHILD;
	}

en ElementChild een getKind() geeft, dan kan je in FrankElement:

	private LinkedHashMap<String, FrankAttribute> attributes;
	private LinkedHashMap<ConfigChild.Key, ConfigChild> configChildren;

vervangen door:

	private Map<Kind, LinkedHashMap<Object, ElementChild>> children;
	
Je kan dan heel veel code die je nu apart hebt uitgeschreven voor Attributes en ConfigChildren op het niveau van ElementChild brengen.
Bijvoorbeeld:

	FrankAttribute findAttributeMatch(FrankAttribute attribute) {
		return attributes.get(attribute.getKey());
	}

	ConfigChild findConfigChildMatch(ConfigChild configChild) {
		return configChildren.get(configChild.getKey());
	}
	
 vervangen door
 
 	<C extends ElementChild> C findElementChild(C child) {
		return children.get(child.getKind()).get(child.getKey());
	}

Je hoeft dan ook geen ``elementChild instanceof`` meer te doen.

Reply:
Gelukt! Dank voor deze tip!